src/Physics/QuasiElastic/EventGen/QELEventGeneratorSM.cxx

```diff
@@ -451,374 +456,289 @@ void QELEventGeneratorSM::LoadConfig(void)
-  if(xsec_max>0) return xsec_max;
-
-  LOG("Kinematics", pINFO)
-                  << "Attempting to compute the max{dxsec/dK} value";
-  xsec_max = this->ComputeMaxXSec2(interaction);
-  if(xsec_max>0) {
-     LOG("Kinematics", pINFO) << "max{dxsec/dK} = " << xsec_max;
-     this->CacheMaxXSec2(interaction, xsec_max);
+  switch (nkey){
+  case 0:
+     return this->ComputeMaxXSec(interaction);
+  
+  case 1:
+  {
+     Range1D_t rQ2 = sm_utils->Q2QES_SM_lim();
+     if (rQ2.max<fQ2Min)
+     {
+        return -1.;
+     }
+     double xsec_max = -1;
+     double tmp_xsec_max = -1;
+     double Q20, v0;
+     const int N_Q2 = 32;
+     const InitialState & init_state = interaction -> InitState();
+     Target * tgt = init_state.TgtPtr();
+     bool isHeavyNucleus = tgt->A()>3;
+     int N_v = isHeavyNucleus?32:0;
+     
+     const double logQ2min = TMath::Log(fQ2Min);
+     const double logQ2max = TMath::Log(rQ2.max);
+     Kinematics * kinematics = interaction->KinePtr();
+     const double pFmax = sm_utils->GetFermiMomentum();
+     // Now scan through kinematical variables Q2,v,kF
+     for (int Q2_n=0; Q2_n <= N_Q2; Q2_n++)
+     {
+        // Scan around Q2
+        double Q2 = TMath::Exp(Q2_n*(logQ2max-logQ2min)/N_Q2 + logQ2min);
+        kinematics->SetKV(kKVQ2, Q2);
+        Range1D_t rv  = sm_utils->vQES_SM_lim(Q2);
+        const double logvmin = TMath::Log(TMath::Max(rv.min, eps));
+        const double logvmax = TMath::Log(TMath::Max(rv.max, TMath::Max(rv.min, eps)));
+        for (int v_n=0; v_n <= N_v; v_n++)
+        {
+           // Scan around v
+           double v = TMath::Exp(v_n*(logvmax-logvmin)/N_v + logvmin);
+           kinematics->SetKV(kKVv, v);
+           kinematics->SetKV(kKVPn, pFmax);
+           // Compute the QE cross section for the current kinematics
+           double xs = fXSecModel->XSec(interaction, fkps);
+           if (xs > tmp_xsec_max)
+           {
+              tmp_xsec_max = xs;
+              Q20 = Q2;
+              v0 =  v;
+           }
+        } // Done with v scan
+     }// Done with Q2 scan
+     
+     const double Q2min = fQ2Min;
+     const double Q2max = rQ2.max;
+     const double vmin = sm_utils->vQES_SM_min(Q2min, Q2max);
+     const double vmax = sm_utils->vQES_SM_max(Q2min, Q2max);
+     ROOT::Math::Minimizer * min = ROOT::Math::Factory::CreateMinimizer("Minuit", "Minimize");
+     ROOT::Math::IBaseFunctionMultiDim * f = isHeavyNucleus?static_cast<ROOT::Math::IBaseFunctionMultiDim*>(new d3XSecSM_dQ2dvdkF_E(fXSecModel, interaction, pFmax)):
+                                                            static_cast<ROOT::Math::IBaseFunctionMultiDim*>(new d1XSecSM_dQ2_E(fXSecModel, interaction));
+     min->SetFunction( *f );
+     min->SetMaxFunctionCalls(10000);  // for Minuit/Minuit2
+     min->SetMaxIterations(10000);     // for GSL
+     min->SetTolerance(0.001);
+     min->SetPrintLevel(0);
+     double step = 1e-7;
+     min->SetVariable(0, "Q2", Q20, step);
```

Comment:
are these parameters (e.g. `MaxFunctionCalls`, `MaxIterations`, `Tolerance` etc) something that should be exposed as algorithm parameters to allow the user to modify them at run time?